src/main/java/com/hcl/appscan/jenkins/plugin/actions/ResultsRetriever.java

```diff
@@ -66,18 +70,38 @@ public boolean getHasResults() {
 	}
 	
 	public boolean checkResults(Run<?,?> r) {
-		if(r.getAllActions().contains(this) && m_provider.hasResults()) {
-			r.getActions().remove(this); //We need to remove this action from the build, but getAllActions() returns a read-only list.
-			r.addAction(createResults());
+		boolean results = false;
+		if (futureTask != null && futureTask.isDone()) {
 			try {
-				r.save();
-			} catch (IOException e) {
+				results = futureTask.get();
+			} catch (Exception e) {
```

Comment:
@mattmurp & @tanmoyguha-hcl

You sure you don't want to log that or something?  Jenkins has a nice logging system for things like that. (And yeah, I know this is inherited code 😜).

Reply:
Yup, that should definitely be logged at a minimum :).  We'll get that taken care of.